### ISSUE TITLE
[Types] Errors occur after 'tsc --noEmit'

### ISSUE THREAD

**Issue by vitorbarbosa19** [anthropics/anthropic-sdk-typescript#449]
After updating to `v0.24.1` from `v0.20.8`, I'm getting two type errors when I run `tsc --noEmit` which is part of my build.

The type errors are:

```
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:19:21 - error TS2345:
Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.
```

And

```
node_modules/@anthropic-ai/sdk/src/lib/MessageStream.ts:14:27 - error TS2307:
Cannot find module '../resources.js' or its corresponding type declarations.
```

These errors are coming straight from anthropic's lib

**Comment by FlowerWrong** [comment]
The same errors.


```
yarn run v1.22.22
$ yarn env:esm nest start
$ cross-env NODE_OPTIONS=--experimental-specifier-resolution=node nest start
(node:27549) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
(Use `node --trace-warnings ...` to show where the warning was created)
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:19:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

19         tokens.push({
                       ~
20           type: 'brace',
   ~~~~~~~~~~~~~~~~~~~~~~~~
21           value: '{',
   ~~~~~~~~~~~~~~~~~~~~~
22         });
   ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:29:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

29         tokens.push({
                       ~
30           type: 'brace',
   ~~~~~~~~~~~~~~~~~~~~~~~~
31           value: '}',
   ~~~~~~~~~~~~~~~~~~~~~
32         });
   ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:39:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

39         tokens.push({
                       ~
40           type: 'paren',
   ~~~~~~~~~~~~~~~~~~~~~~~~
41           value: '[',
   ~~~~~~~~~~~~~~~~~~~~~
42         });
   ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:49:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

49         tokens.push({
                       ~
50           type: 'paren',
   ~~~~~~~~~~~~~~~~~~~~~~~~
51           value: ']',
   ~~~~~~~~~~~~~~~~~~~~~
52         });
   ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:59:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

59         tokens.push({
                       ~
60           type: 'separator',
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
61           value: ':',
   ~~~~~~~~~~~~~~~~~~~~~
62         });
   ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:69:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

69         tokens.push({
                       ~
70           type: 'delimiter',
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
71           value: ',',
   ~~~~~~~~~~~~~~~~~~~~~
72         });
   ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:107:23 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

107           tokens.push({
                          ~
108             type: 'string',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
109             value,
    ~~~~~~~~~~~~~~~~~~
110           });
    ~~~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:135:21 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

135         tokens.push({
                        ~
136           type: 'number',
    ~~~~~~~~~~~~~~~~~~~~~~~~~
137           value,
    ~~~~~~~~~~~~~~~~
138         });
    ~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts:155:23 - error TS2345: Argument of type '{ type: string; value: string; }' is not assignable to parameter of type 'never'.

155           tokens.push({
                          ~
156             type: 'name',
    ~~~~~~~~~~~~~~~~~~~~~~~~~
157             value,
    ~~~~~~~~~~~~~~~~~~
158           });
    ~~~~~~~~~~~
node_modules/@anthropic-ai/sdk/src/lib/MessageStream.ts:14:27 - error TS2307: Cannot find module '../resources.js' or its corresponding type declarations.

14 import { TextBlock } from "../resources.js";
                             ~~~~~~~~~~~~~~~~~

Found 10 error(s).
```

**Comment by FlowerWrong** [comment]
I patched it.

```
diff --git a/node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts b/node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts
index b79cdef..8b4b921 100644
--- a/node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts
+++ b/node_modules/@anthropic-ai/sdk/src/_vendor/partial-json-parser/parser.ts
@@ -5,7 +5,7 @@ type Token = {
 
 const tokenize = (input: string) => {
     let current = 0;
-    let tokens = [];
+    let tokens: any[] = [];
 
     while (current < input.length) {
       let char = input[current];
diff --git a/node_modules/@anthropic-ai/sdk/src/lib/MessageStream.ts b/node_modules/@anthropic-ai/sdk/src/lib/MessageStream.ts
index 428047e..6b77bd8 100644
--- a/node_modules/@anthropic-ai/sdk/src/lib/MessageStream.ts
+++ b/node_modules/@anthropic-ai/sdk/src/lib/MessageStream.ts
@@ -8,10 +8,10 @@ import {
   MessageParam,
   MessageCreateParams,
   MessageCreateParamsBase,
+  TextBlock,
 } from "../resources/messages.js";
 import { type ReadableStream } from "../_shims/index.js";
 import { Stream } from "../streaming.js";
-import { TextBlock } from "../resources.js";
 import { partialParse } from "../_vendor/partial-json-parser/parser.js";
 
 export interface MessageStreamEvents {

```

**Comment by RobertCraigie** [comment]
This should be fixed in the next release: https://github.com/anthropics/anthropic-sdk-typescript/pull/456

**Comment by RobertCraigie** [comment]
This fix has been released in `v0.24.3`
